brats/transformations.py

```diff
@@ -73,3 +78,77 @@ def hist_match(source: np.ndarray, template: np.ndarray) -> np.ndarray:
                                     template_values)
         result[positive] = interp_t_values[bin_idx]
         return result
+
+
+class NiftiOrderTransformation:
+    """
+    Changes dimensions order from nifti (H,W,D,C) to torch convention (C,D,W,H).
+    It can be applied to both ``torch.Tensor`` or ``numpy.ndarray``.
+    """
+
+    def __call__(self, img: typing.Union[np.ndarray, torch.Tensor]) -> typing.Union[np.ndarray, torch.Tensor]:
+        assert len(img.shape) == 4
+
+        if isinstance(img, torch.Tensor):
+            transformed = img.permute(3, 2, 0, 1)
+        if isinstance(img, np.ndarray):
+            transformed = np.moveaxis(img, [0, 1, 2, 3], [2, 3, 1, 0])
+        return transformed
+
+
+class AddChannelDimToMaskTransformation:
+    """
+    Adds additional channel dimension
+    It can be applied to both ``torch.Tensor`` or ``numpy.ndarray``.
+    """
+
+    def __call__(self, img: typing.Union[np.ndarray, torch.Tensor]) -> typing.Union[np.ndarray, torch.Tensor]:
+        assert len(img.shape) == 3
+
+        if isinstance(img, torch.Tensor):
+            transformed = torch.unsqueeze(img, 3)
+        if isinstance(img, np.ndarray):
+            transformed = np.expand_dims(img, 3)
+        return transformed
+
+
+class BinarizationTransformation:
+    """
+    Adds additional channel dimension
+    It can be applied to``numpy.ndarray``.
+    """
+
+    def __call__(self, img: typing.Union[np.ndarray, torch.Tensor]) -> typing.Union[np.ndarray, torch.Tensor]:
+        img[img > 0] = 1
```

Comment:
I think this modifies argument in-place.

Reply:
It does. I used `singledispatch` in order to utilize modules copying functionalities, especially
`clone()` method from `torch.Tensor`. By default is uses `copy` module.

I'm not sure if it is not an overkill, if you have a better solution, comment.